pkg/ebpf/probes/probes.go

```diff
@@ -492,6 +507,83 @@ func (p *tcProbe) autoload(module *bpf.Module, autoload bool) error {
 	return enableDisableAutoload(module, p.programName, autoload)
 }
 
+//
+// Cgroup
+//
+
+type cgroupProbe struct {
+	programName string
+	attachType  bpf.BPFAttachType
+	bpfLink     *bpf.BPFLink
+}
+
+// attach attaches an eBPF program to a cgroup
+func (p *cgroupProbe) attach(module *bpf.Module, args ...interface{}) error {
+	var link *bpf.BPFLink
+
+	if p.bpfLink != nil {
+		return nil // already attached, it is ok to call attach again
+	}
+
+	// cgroup attachments root directory
+
+	cgroupRootDir := "/sys/fs/cgroup/unified"
+	if _, err := os.Stat(cgroupRootDir); os.IsNotExist(err) {
+		cgroupRootDir = "/sys/fs/cgroup"
+	}
+
+	// TODO: mount cgroupv2 filesystem and use the mp for attaching (rhel8)
```

Comment:
I think we should always mount cgroupv2 (when possible) instead of looking for its path

Reply:
Yes, this is a problem for Ubuntu Focal and RHEL8 clones (mostly RHEL8 clones which don't have cgroupv2 mounted by default, although they do support it). I'll mount cgroupv2 during attachment if not mounted already (and push changes).

Reply:
@yanivagman I have started https://github.com/aquasecurity/tracee/pull/2233 and I think we should merge this "as is" (Since it works) and review that PR (after some fixes of mine) and merge that one to solve this issue (since its a refactor for how we deal with cgroup mounting/info).